core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2393,6 +2407,10 @@ class Log(@volatile private var _dir: File,
       }
       // okay we are safe now, remove the swap suffix
       sortedNewSegments.foreach(_.changeFileSuffixes(Log.SwapFileSuffix, ""))
+
+      // If not recovered swap file we need to increment logStartOffset here. Otherwise, we do this when loading the log.
+      if (!isRecoveredSwapFile)
+         maybeIncrementLogStartOffset(segments.firstEntry.getValue.baseOffset, LogCompaction)
```

Comment:
Is this referring to the comment about excluding recovered swap files, or is there an issue that will occur with segment splitting?

Reply:
What I am referring to is the use of the `LogCompaction` reason passed to `maybeIncrementLogStartOffset`. I believe this logic also applies when splitting a segment and that does not involve compaction.

Reply:
Ah. I see what you mean. I can add such a reason as well as a parameter for this method. In the case of `completeSwapOperations` I'm a little unsure how we can tell. We can check the number of swap files since cleaning will always make just 1. But I think in the case of splitting, we may also have just 1 if we fail right before handling the last swap file. This seems like a pretty rare case though.

Reply:
Decided to rename the reason to generalize.